src/NuGet.Core/NuGet.Protocol/Providers/ServiceIndexResourceV3Provider.cs

```diff
@@ -131,6 +125,8 @@ private async Task<ServiceIndexResourceV3> GetServiceIndexResourceV3(
 
             for (var retry = 1; retry <= maxRetries; retry++)
             {
+                token.ThrowIfCancellationRequested();
```

Comment:
There are only two methods that might take some time, `client.GetAsync` and `Task.Delay`, but the cancellation token is already passed in both places.  So, I don't imagine that this `token.ThrowIf..` will ever prevent save any significant CPU time